### PR TITLE
Confirm or stop deleting a song

### PR DESCRIPTION
**Pull request by ninapeters**
Hello! I've added logic that you can use to confirm or stop deleting a song. I also found and fixed a bug: by clicking the delete button, the song stops playing if it was currently active. I'm looking forward to your feedback! ✨

### CODE REVIEW

File: src/components/SavedSongs/SavedSong.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  deleteSavedSong,
+  toggleCurrentSongId,
+  isSongPlaying,
+  currentSongId,
+}) {
+  const [toBeDeleted, setToBeDeleted] = useState(false)
+
+  return (
+    <>
+      {!toBeDeleted && (
```
Comment (jamarob) [inline src/components/SavedSongs/SavedSong.js:35]: nit: or `toBeDeleted || (..`

File: src/components/SavedSongs/SavedSong.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
+import styled from 'styled-components/macro'
+import PropTypes from 'prop-types'
+import { useState } from 'react'
+import AudioButton from '../Buttons/AudioButton'
+import ButtonSecondary from '../Buttons/ButtonSecondary'
+import { ReactComponent as XIcon } from '../../assets/x.svg'
+import { ReactComponent as ResetIcon } from '../../assets/reset.svg'
+import { ReactComponent as BinIcon } from '../../assets/bin.svg'
```
Comment (jamarob) [inline src/components/SavedSongs/SavedSong.js:8]: Maybe introduce an icons file from which you reexport your assets?

File: src/components/SavedSongs/SavedSong.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            onClick={() => setToBeDeleted(false)}
+            data-testid="reset-button"
+          >
+            <ResetIcon />
+          </ConfirmationButton>
+        </ConfirmationWrapper>
+      )}
+    </>
+  )
+  function prepareDeletion() {
```
Comment (jamarob) [inline src/components/SavedSongs/SavedSong.js:78]: nit: add a line break ;)

File: src/components/SavedSongs/SavedSong.test.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        id="a"
+        deleteSavedSong={() => {}}
+        toggleCurrentSongId={() => {}}
+        isSongPlaying={false}
+        currentSongId="a"
+      />
+    )
+    expect(container.firstChild).toMatchSnapshot()
+  })
+  it('calls toggleCurrentSongId with the correct id', () => {
```
Comment (jamarob) [inline src/components/SavedSongs/SavedSong.test.js:20]: nit: separate tests with  newline for readability

File: src/components/SavedSongs/SavedSong.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+  return (
+    <>
+      {!toBeDeleted && (
+        <ListItem>
+          <ButtonWrapper>
+            <ButtonSecondary
+              id={id}
+              onClick={prepareDeletion}
+              data-testid="x-button"
```
Comment (jamarob) [inline src/components/SavedSongs/SavedSong.js:41]: Maybe `prepare-delete-button` as a name?

File: src/components/SavedSongs/SavedSong.test.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        currentSongId="a"
+      />
+    )
+    const Button = getByTestId('x-button')
+    user.click(Button)
+    expect(getByText('Lorem ipsum')).toBeInTheDocument()
+    expect(getByTestId('reset-button')).toBeInTheDocument()
+    expect(getByTestId('delete-button')).toBeInTheDocument()
+  })
+  it('toggles the isSongPlaying state correctly', () => {
```
Comment (jamarob) [inline src/components/SavedSongs/SavedSong.test.js:70]: Name of the test indicates there is some kind of user action to toggle the state, but you simply render it with different props. Maybe rename the test?

File: src/components/SavedSongs/SavedSongList.js [hunk 7]
```diff
@@ -21,26 +22,17 @@ export default function SavedSongList({
   return (
     <ListStyled>
```
Comment (jamarob) [inline src/components/SavedSongs/SavedSongList.js:23]: You habe no naming conflict, so the `Styled` postfix is not needed, maybe simply call it `List` or `SongsList` for expressiveness?

File: src/components/SavedSongs/SavedSongList.md [hunk 8]
```diff
@@ -1,14 +1,21 @@
-One saved song item:
+List of saved songs:
 
 ```js
 <SavedSongList
+  deleteSavedSong={() => {}}
```
Comment (jamarob) [inline src/components/SavedSongs/SavedSongList.md:5]: nit: `() => {}` is named noop by convention

**Review by jamarob — COMMENTED** [review]
Cleanly written and easy to follow, well done! 🎉 
just left some nit picky comments

File: src/components/SavedSongs/SavedSong.js (hunk 5 above)
Comment (dajanaheinze) [reply src/components/SavedSongs/SavedSong.js:41]: I agree this, or is this button used for something else?

**Review by dajanaheinze — COMMENTED** [review]
great, just a little thing :-)

File: src/hooks/useAudio.js [hunk 9]
```diff
@@ -41,9 +41,15 @@ export default function useAudio({ artistData }) {
     }
   }
 
+  function stopPlayingSong() {
+    songRef.current.stop()
+    setIsSongPlaying(false)
+  }
+
```
[the remark this answers is not in the document]
Comment (ninapeters) [reply src/hooks/useAudio.js:48]: In that case, I really want to stop the song so there aren't any leftovers. When pausing, the current seek position is saved.

File: src/components/SavedSongs/SavedSong.js (hunk 5 above)
Comment (ninapeters) [reply src/components/SavedSongs/SavedSong.js:41]: Good point! Thanks :)

**Pull request merged**